ProcessMaker/processmaker
# Ordering the processes correctly when starting the process - PM4EH-280

## Issue & Reproduction Steps
There is a sorting error when multiple categories (`category_assignments` table) are associated with a process.

This happens because it is ordered by the associated category in the `processes.process_category_id` column.

Unable to sort correctly on front-end because of pagination which brings partial results.

## Solution
- Change the query to bring the related category in `category_assignments` and order it correctly.
- **Do not change the default API output**, to **maintain backwards compatibility** and not cause unexpected bugs.

[PM4EH-280-sorting-issue.webm](https://user-images.githubusercontent.com/16992680/214589081-447bd1f4-5464-4623-919d-2cc039b048fc.webm)

## How to Test
- Create a category "1", "A", "B"
- Create processes with these names for category "A": "A1", "A2", "A3", "A4", "A5", "A6", "A7", "A8"
- Create processes with these names for category "B": "B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8"
- Associate a second category "1" to the processes: "A4", "A6", "B5", "B1"
- The first page should contain category "1" with the 4 related processes.

## Related Tickets & Packages
- [PM4EH-280](https://processmaker.atlassian.net/browse/PM4EH-280)
- [FOUR-7310](https://processmaker.atlassian.net/browse/FOUR-7310)

## Code Review Checklist


[PM4EH-280]: https://processmaker.atlassian.net/browse/PM4EH-280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


.
